src/transaction.js

```diff
@@ -250,8 +399,22 @@ Transaction.prototype.getId = function () {
   return [].reverse.call(this.getHash()).toString('hex')
 }
 
-Transaction.prototype.toBuffer = function () {
-  var buffer = new Buffer(this.byteLength())
+Transaction.prototype.toBuffer = function (withWitness) {
```

Comment:
I think it would make more sense that `withWitness` would default to TRUE, because the only reason for wanting the non-witness is to `getId` or if you want to submit it to an old node or some old piece of software

and in the tests that would mean I'll add a `txHexWithoutWitness` instead of the `txHexWithWitness` that I've added so far in a few places ... 

/discuss